hr_holidays_hour/models/resource_calendar.py

```diff
@@ -0,0 +1,144 @@
+        assert start_dt.date() == end_dt.date(), \
+            'get_working_intervals_of_day is restricted to one day'
+        return start_dt, work_limits
+
+    @api.multi
+    def _get_intervals(self, start_dt, work_dt, work_limits):
+
+        def get_interval(work_day_hour, work_dt):
+            interval = work_dt.replace(
+                hour=int(work_day_hour),
+                minute=int(
+                    (work_day_hour - int(work_day_hour)) * 60
+                )
+            )
+            return interval
+
+        work_intervals = []
+        for work_day in self.get_attendances_for_weekday(start_dt):
+            interval_start = get_interval(work_day.hour_from, work_dt)
+            interval_stop = get_interval(work_day.hour_to, work_dt)
+
+            working_interval = (interval_start, interval_stop)
+            work_intervals += self.interval_remove_leaves(
+                working_interval,
+                work_limits
+            )
+        return work_intervals
+
+    @api.multi
+    def get_working_intervals_of_day(
+            self, start_dt=None, end_dt=None,
+            leaves=None, compute_leaves=False, resource_id=None,
+            default_interval=None):
+        """ Override method because of the working intervals not
+        calculating the minutes only the hours"""
+
+        def working_interval(default_interval, start_dt):
+            working_interval = []
+            if default_interval:
+                working_interval = (
+                    start_dt.replace(
+                        hour=default_interval[0], minute=0, second=0,
+                        microsecond=0),
+                    start_dt.replace(
+                        hour=default_interval[1], minute=0, second=0,
+                        microsecond=0))
+            return working_interval
+
+        start_dt, work_limits = self._get_work_limits(end_dt, start_dt)
+
+        work_dt = start_dt.replace(hour=0, minute=0, second=0)
+
+        # no calendar: try to use the default_interval, then return directly
+        if not self:
+            working_interval = working_interval(default_interval, start_dt)
+            intervals = self.interval_remove_leaves(
+                working_interval,
+                work_limits
+            )
+            return intervals
+
+        # find leave intervals
+        work_intervals = self._get_intervals(start_dt, work_dt, work_limits)
+
+        if not leaves and compute_leaves:
+            leaves = self.get_leave_intervals(resource_id=resource_id)
+
+        # filter according to leaves
+        intervals = []
+        for interval in work_intervals:
+            work_interval = self.with_context(
+                change_tz=True
+            ).interval_remove_leaves(interval, leaves)
+            intervals += work_interval
+
+        return intervals
+
+    @api.model
+    def interval_remove_leaves(self, interval, leave_intervals):
```

Comment:
if we need this, I think there's something wrong with UTC handling at the place the produces the leaves